api/RESToverHTTP/GET-rfq-comprehensive.py

```diff
@@ -0,0 +1,186 @@
+                                                       method=method)
+
+        # HTTP Request Headers
+        headers = {
+                    'Paradigm-API-Timestamp': timestamp,
+                    'Paradigm-API-Signature': signature,
+                    'Authorization': f'Bearer {self.access_key}'
+                    }
+
+        # GET Request
+        response = requests.get(urljoin(self.host, action),
+                                headers=headers)
+
+        # Check response code for error
+        if response.status_code != 200 and response.status_code != 201:
+            print('Incorrect response code: {0}'.format(response.status_code))
+
+        print('Response Status Code: {}'.format(response.status_code))
+        print('Response Text:')
+        print(response.text)
+
+        # To paginate the returned results and ensure all results are returned
+        page_check = json.loads(response.text)['next']
+
+        if page_check is not None:
+            if page_check > self.page:
+                self.page = page_check
+                self.get_rfq()
+
+    def generate_signature(self, action, method):
+        message = method.encode('utf-8') + b'\n'
+        message += action.encode('utf-8') + b'\n'
+
+        timestamp = str(int(time.time() * 1000))
+        message = timestamp.encode('utf-8') + b'\n' + message
+        digest = hmac.digest(self.signing_key, message, 'sha256')
+        signature = base64.b64encode(digest)
+
+        return signature, timestamp
+
+    def get_rfq(self):
+        return self.request(action='/rfq/', method='GET')
+
+
+if __name__ == "__main__":
+    # Arguments
+    my_parser = argparse.ArgumentParser()
+
+    my_parser.add_argument('RFQ_ID',
+                        metavar='rfq_id',
+                        type=str,
+                        help='RFQ Id',
+                        default="-1",
+                        nargs='?')
+    my_parser.add_argument('CLIENT_ORDER_ID',
+                        metavar='client_order_id',
+                        type=str,
+                        help='Client Order Id',
+                        default="-1",
+                        nargs='?')
+    my_parser.add_argument('PAGE',
+                        metavar='page',
+                        type=str,
+                        help='Page Number of returned results',
+                        default="-1",
+                        nargs='?')
+    my_parser.add_argument('STATUS',
+                        metavar='status',
+                        type=str,
+                        help='Status of RFQ ACTIVE|CANCELED',
+                        default="-1",
+                        nargs='?')
+    
+    my_parser.add_argument('ACCESS_KEY',
+                        metavar='api_key',
+                        type=str,
+                        help='Paradigm Account Access Key',
+                        default='kNLMKQnbcM0xGSL2wOntLwpy',
```

Comment:
Might be better to remove this as a default and avoid having static keys in here.